src/coreclr/gc/gc.cpp

```diff
@@ -1385,11 +1473,12 @@ static void enter_spin_lock_noinstru (RAW_KEYWORD(volatile) int32_t* lock)
 {
 retry:
 
-    if (Interlocked::CompareExchange(lock, 0, -1) >= 0)
+    if (Interlocked::CompareExchange(lock, lock_taken, lock_free) != lock_free)
```

Comment:
this method doesn't actually need to change, correct? it will work as is. but I do think it's better to use the actualy enum instead of an integer. perhaps add a comment to say that this will never be used by the more space lock?

Reply:
Right, this doesn't need a change, I just did it for clarity. Will add a comment as suggested.